sqlglot/dialects/snowflake.py

```diff
@@ -585,16 +585,21 @@ def _parse_at_before(self, table: exp.Table) -> exp.Table:
 
             return table
 
-        def _parse_table_parts(
-            self, schema: bool = False, is_db_reference: bool = False, wildcard: bool = False
-        ) -> exp.Table:
+        def _parse_file_location(self) -> t.Optional[exp.Expression]:
             # https://docs.snowflake.com/en/user-guide/querying-stage
             if self._match(TokenType.STRING, advance=False):
-                table = self._parse_string()
+                file = self._parse_string()
             elif self._match_text_seq("@", advance=False):
-                table = self._parse_location_path()
+                file = self._parse_location_path()
             else:
-                table = None
+                file = None
+
+            return file
+
+        def _parse_table_parts(
+            self, schema: bool = False, is_db_reference: bool = False, wildcard: bool = False
+        ) -> exp.Table:
+            table = self._parse_file_location()
```

Comment:
It would be ideal if we could avoid the method call overhead here because `_parse_table_parts` is hot. This may be doable if we implement the suggestion re: overriding in T-SQL and keeping it local to the `COPY` statement.